### ISSUE TITLE
Question : How to run installed apps using command prompt?

### ISSUE THREAD

**Issue by atifwaqar** [Starcounter/Starcounter#3889]
I have 3 StarCounter apps installed :
1. UserAdmin
2. SignIn
3. DatabaseEditor
![image](https://cloud.githubusercontent.com/assets/4388707/19432095/4deb75f6-945b-11e6-99e1-ecedb97fb147.png)

1. How can I run these using a command in command prompt?
2. Is there any app which monitors these apps and run them if they are not running?

**Comment by per-samuelsson** [comment]
cc @andwah

**Comment by andwah** [comment]
> How can I run these using a command in command prompt?

We do not have a command line tool for managing the Administrator.  the closest thing is to use `star.exe` or `staradmin.exe`.  Installed apps exist in the database repository on your drive. So you may use `star.exe` to run an installed executable.

> Is there any app which monitors these apps and run them if they are not running?

No.

**Comment by atifwaqar** [comment]
> So you may use star.exe to run an installed executable.

Is there any document on how to use `star.exe` to run an installed executable?

**Comment by andwah** [comment]
> Is there any document on how to use star.exe to run an installed executable?

No. it's more like a workaround to solve your issue (short term).  

Maybe we should extend `star.exe` to support the handling of `installed` applications or in some other way  be able to do this from command line.

**Comment by atifwaqar** [comment]
I have installed 3 apps from SC App Warehouse : 
![image](https://cloud.githubusercontent.com/assets/4388707/19538735/c3cc848a-9655-11e6-85a8-a78e9adb1a49.png)

When I click on "Start" database. All the installed apps are started automatically. (So far everything is perfect)
![image](https://cloud.githubusercontent.com/assets/4388707/19538892/9b4ccd98-9656-11e6-8e42-62fd4806ae77.png)

Now, I stop my database and use `staradmin start db` command to start my database, only database is started, but none of the apps are started.
![image](https://cloud.githubusercontent.com/assets/4388707/19538841/60867812-9656-11e6-909d-fe42ab3e9d06.png)

Is there an easy way to trigger the DB Start Button function behind `staradmin start db` command or anyother command?
I have tried both commands : 
```
staradmin start server
staradmin start db
```

**Comment by dan31** [comment]
@andwah: is @atifwaqar question about playlists?

**Comment by per-samuelsson** [comment]
@dan31 

> @andwah: is @atifwaqar question about playlists?

Not exactly. (And btw, we should use "app suites" rather than playlists).

What we lack is support for all management of "installed applications" - i.e applications being downloaded and installed from some store/warehouse - in our *core* management tools. The admin server library and our CLI tooling like `star` and `staradmin` simply don't know about that concept at all. It's built completely as a layer on top of core management, introducing separate tool `starpack` + functionality supported only in `administrator.exe`, not the core library (which is the real source and guard of all database- and app-management state).

[Here's the proposal](https://github.com/Starcounter/RebelsLounge/issues/89) I've made to get that whole project started.

**Comment by andwah** [comment]
> @andwah: is @atifwaqar question about playlists?

No. it's about starting apps that has been downloaded and unpacked with the administrator.
That apps can be started with `star.exe` as normal but the tricky part is to find where the apps was unpacked (they are unpacked in the database repository).

> (And btw, we should use "app suites" rather than playlists).

**App suite** : is a packet of apps,
**Playlist** : is a list of apps/apps suits that should start when the database is started.
 
Im in working on an assambly that can handle "installed" apps it will be used for the "hosting" project that im working on, maybe this assembly also can be used by the "core" ?

**Comment by per-samuelsson** [comment]
> **App suite** : is a packet of apps,
> **Playlist** : is a list of apps/apps suits that should start when the database is started.

Ah, I was wrong then. Thanks for explaining. That make sense. The I guess playlist have similarities to proposed "system apps" (i.e. the start with the first host / database).

> Im in working on an assambly that can handle "installed" apps it will be used for the "hosting" project that im working on, maybe this assembly also can be used by the "core" ?

Hard to say, but maybe, yes. No matter how, what we should strive for is a single, well-working management model - not have it in different places. I know we both agree on that, just that there is never any real time to synchronize work on that.
